### PR TITLE
Add operator online duration endpoint

### PR DESCRIPTION
**Pull request by indrekj**
Ref [#106621868]

Related: https://github.com/salemove/web_services/pull/590

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -1430,3 +1430,44 @@ Possible dimensions that can be used in filters:
...
         }
     ]
+
+### Operator Online Duration [POST /operators/stats/online_duration]
+Returns operator online duration in seconds.
+
+Possible dimensions that can be used in filters:
+* `site_id` - Site ID
+* `operator_id` - Operator ID
+* `type` - Operator online status. There's three group of types an online operator can have:
```
Comment (ostap0207) [inline apiary.apib]: Small mistake "There're three groups"

Comment (indrekj) [inline apiary.apib]: http://english.stackexchange.com/questions/12865/is-therere-similar-to-theres-a-correct-contraction 
Used the long form instead

File: apiary.apib [hunk 2]
```diff
@@ -1430,3 +1430,44 @@ Possible dimensions that can be used in filters:
...
     ]
+
+### Operator Online Duration [POST /operators/stats/online_duration]
+Returns operator online duration in seconds.
+
+Possible dimensions that can be used in filters:
+* `site_id` - Site ID
+* `operator_id` - Operator ID
+* `type` - Operator online status. There are three group of types an online operator can have:
+  * Available media type if the operator is available - `video`, `audio`, `chat`
```
Comment (ostap0207) [inline apiary.apib]: Forgot about ```group -> groups```

Comment (ostap0207) [inline apiary.apib]: Otherwise +1

**Comment by indrekj** [comment]
Added [on hold] for now. Currently type is `text` instead of `chat`. I need to figure out how to rename it to chat. I could make it properly and rename it everywhere, but this is not very easy. I could also just add some renamer to the statistics service. This would make it easier but requires additional processors.

**Comment by deiwin** [comment]
Did salemove/message-hub#283 fix this? or is there still work to be done?

File: apiary.apib [hunk 3]
```diff
@@ -1430,3 +1430,44 @@ Possible dimensions that can be used in filters:
             "timestamp": "2015-09-02T00:00:00.000Z",
         }
     ]
+
+### Operator Online Duration [POST /operators/stats/online_duration]
```
Comment (deiwin) [inline apiary.apib]: Wait, why are these `POST`s?

File: apiary.apib [hunk 4]
```diff
@@ -1430,3 +1430,44 @@ Possible dimensions that can be used in filters:
...
+Returns operator online duration in seconds.
+
+Possible dimensions that can be used in filters:
+* `site_id` - Site ID
+* `operator_id` - Operator ID
+* `type` - Operator online status. There are three groups of types an online operator can have:
+  * Available media type if the operator is available - `video`, `audio`, `chat`
+  * `engaged` if the operator is currently in an engagement
+  * An unavailable type if the operator is marked away or busy - `Quick Break`, `Training`, `With Customer`, etc
```
Comment (deiwin) [inline apiary.apib]: Do we really want to release this ... ?

Comment (deiwin) [inline apiary.apib]: I mean the whole type thing

Comment (indrekj) [inline apiary.apib]: What do you mean?

**Comment by indrekj** [comment]
Yes, the mh PR fixed it. Well, I have to recalculate the segments after the deploy

**Comment by deiwin** [comment]
We have a separate release branch now, so we can merge these without the changes having to be live

**Comment by deiwin** [comment]
+1

File: apiary.apib (hunk 3 above)
Comment (indrekj) [inline apiary.apib]: GET does not support json payloads (nicely).

**Comment by stenver** [comment]
!squash

**Pull request merged**